### ISSUE TITLE
classification_losses

### ISSUE THREAD

**Issue by Hesene** [zylo117/Yet-Another-EfficientDet-Pytorch#539]

**Comment by rvandeghen** [comment]
Hi,
this comes from the original paper: _Focal Loss for Dense Object Detection_ as stated below:
> The total focal loss of an image is computed as the sum of the focal loss over all ∼100k anchors, _normalized by the number of anchors assigned to a ground-truth box_.

**Comment by Hesene** [comment]
> Hi,
> this comes from the original paper: _Focal Loss for Dense Object Detection_ as stated below:
> 
> > The total focal loss of an image is computed as the sum of the focal loss over all ∼100k anchors, _normalized by the number of anchors assigned to a ground-truth box_.

OK，Thank you very much

**Comment by zylo117** [comment]
Despite of the focal loss part, the `cls_loss` is all about the areas that match with the anchors, and the other areas are zeros.
`cls_loss = torch.where(torch.ne(targets, -1.0), cls_loss, zeros)`

So `cls_loss.sum()` means all cls_losses, including the positive and negative samples' loss within the matched areas and then divide it by `num_positive_anchors` to get the average cls_loss of a single matched anchor.

**Comment by Hesene** [comment]
> Despite the focal loss part, the `cls_loss` is all about the areas that match with the anchors, and the other areas are zeros.
> `cls_loss = torch.where(torch.ne(targets, -1.0), cls_loss, zeros)`
> 
> So `cls_loss.sum()` means all cls_losses, including the positive and negative samples' loss within the matched areas and then divide it by `num_positive_anchors` to get the average cls_loss of a single matched anchor.

Thanks for your sharing, I get it!
